src/Compilers/CSharp/Test/Symbol/Symbols/CovariantReturnTests.cs

```diff
@@ -249,11 +412,13 @@ private static void VerifyAssignments(CSharpCompilation comp, int expectedAssign
             CSharpCompilation comp,
             string assignments,
             MetadataReference[] references = null,
-            TargetFramework targetFramework = TargetFramework.Standard)
+            bool withoutCorlib = false)
         {
             CompilationReference compAsMetadata = comp.ToMetadataReference();
             references = references?.Append(compAsMetadata) ?? new[] { compAsMetadata };
-            var result = CreateCompilation(assignments, references: references, targetFramework: targetFramework);
+            if (!withoutCorlib)
+                references = references.Prepend(comp.References.First()).ToArray();
```

Comment:
>comp.References.First() [](start = 48, length = 23)

It feels like it will be more robust to get a core library symbol from source assembly (we have a property, we might even have a property on compilation) and then locate the reference for it, rather than hardcoding an assumption that core library is always first reference.

Reply:
This isn't really so much an assumption as a fact that is true by construction in the tests where this helper is used. But I agree that would be better.

---
In reply to: [452411411](https://github.com/dotnet/roslyn/pull/45824#discussion_r452411411) [](ancestors = 452411411)